test-main.js

```diff
@@ -0,0 +1,65 @@
+var allTestFiles = ['assert', 'underscore'];
+var TEST_REGEXP = /tests\/.*\.js$/i;
+
+// Automatically clenaup sinon spies and stubs.
+var realSinon = sinon;
+beforeEach(function() {
+    sinon = realSinon.sandbox.create();
+});
+afterEach(function() {
+    sinon.restore();
+    sinon = realSinon;
+});
+
+function withSettings(changes, test) {
```

Comment:
Reloading the settings is presumably by design so there's a new context, but I can see how that would slow things down.

`withSettings` looks really good for the use-cases here and benefits from the lack of new contexts created each time it's run because when settings is already in memory nothing needs to happen.

The only thing I can see that might be missing is a try / finally around the test function so that nothing can prevent the settings teardown from happening. 

Presumably the test func call in between the settings changes is synchronous?